cgi-bin/DW/Controller/Manage/EmailPost.pm

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/perl
+#
+# DW::Controller::Manage::EmailPost
+#
+# This code is based on code originally created by the LiveJournal project
+# owned and operated by Live Journal, Inc. The code has been modified and
+# expanded by Dreamwidth Studios, LLC. These files were originally licensed
+# under the terms of the license supplied by Live Journal, Inc, which made
+# its code repository private in 2014. That license is archived here:
+#
+# https://github.com/apparentlymart/livejournal/blob/master/LICENSE-LiveJournal.txt
+#
+# In accordance with the original license, this code and all its
+# modifications are provided under the GNU General Public License.
+# A copy of that license can be found in the LICENSE file included as
+# part of this distribution.
+#
+# Authors:
+#      Jen Griffin <kareila@livejournal.com>
+#
+# Copyright (c) 2023 by Dreamwidth Studios, LLC.
+#
+
+package DW::Controller::Manage::EmailPost;
+
+use strict;
+
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+use DW::FormErrors;
+
+use LJ::Emailpost::Web;
+
+DW::Routing->register_string( "/manage/emailpost", \&emailpost_handler, app => 1 );
+
+sub emailpost_handler {
+    my $ml_scope = "/manage/emailpost.tt";
+    return error_ml("$ml_scope.error.sitenotconfigured") unless $LJ::EMAIL_POST_DOMAIN;
+
+    my ( $ok, $rv ) = controller( anonymous => 0, form_auth => 1 );
+    return $rv unless $ok;
+
+    my $u = $rv->{u};
+    return DW::Template->render_template( 'error.tt', { message => $LJ::MSG_READONLY_USER } )
```

Comment:
vague thought that is totally out of scope of this PR, I wonder if this should just be part of calling `controller`. Most of the time, we don't want to allow readonly users, particularly if form_auth is 1 and anonymous is 0, so maybe this could be done automatically... I know I never remember to check whether a user is readonly or not, so I bet there's a zillion paths where we don't even try.

Reply:
I can open a new issue for that.